flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/reader/IcebergSourceSplitReader.java

```diff
@@ -46,6 +51,8 @@ class IcebergSourceSplitReader<T> implements SplitReader<RecordAndPosition<T>, I
   private final SerializableComparator<IcebergSourceSplit> splitComparator;
   private final int indexOfSubtask;
   private final Queue<IcebergSourceSplit> splits;
+  private final Set<String> pausedSplits;
```

Comment:
add a comment to explain that synchronization/lock is not needed because all accesses are from single reader thread from `SingleThreadMultiplexSourceReaderBase`

Reply:
Ohh.. I am almost missed this.
Added this comment as well